### DISCUSSION TITLE
Format option in context Menu

### DISCUSSION THREAD

**Discussion by JoaquinFernandez** [logancyang/obsidian-copilot#789] [Ideas]
I regularly have to reformat notes. Sometimes because it is just copying and pasting from a site, sometimes because using chat to generate the response, and everything is ill formatted (not using headers, too much spaces, etc.).

It would be awesome to have a new context menu option:
<img width="341" alt="image" src="https://github.com/user-attachments/assets/79d30e0f-17c7-4f5b-8f38-8cdb923d2240">
That is "Copilot: Format selection". Will fire a specifc MD formatting prompt and output the selected text with the correct expected format.

And a configurable prompt with default.

Would you accept a PR with the changes?

**Comment by logancyang** [comment]
Hi @JoaquinFernandez, thanks for the suggestion. This has been asked for a number of times. My only concern is that it overwhelms the menu, so I added the toggles in Copilot settings for commands in the palette. I think as long as there's a setting where you can enable/disable these commands from the context menu, it's definitely a great addition, and I will accept it.
